### ISSUE TITLE
bug(material/radio): screen readers announce radio buttons within group as "1 of 1" regardless of number

### ISSUE THREAD

**Issue by byanofsky** [angular/components#23580]
#### Reproduction

**Note:** Only reproducible with JAWS and NVDA.

Reproducible within ["Basic radios" demo](https://material.angular.io/components/radio/examples#radio-overview).

Steps to reproduce:
1. Start JAWS or NVDA (screen readers)
1. Tab to a radio button

 
#### Expected Behavior

Announces radio button label and "1 of 2".

#### Actual Behavior

Announces radio button label and "1 of 1".

#### Environment

- Angular: 12.2.5
- CDK/Material: 12.2.5
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by shb98** [comment]
@jelbourn , quick question - what does the label `Fixed in MDC` mean? It's great to see the Pull Request for the fix. But it stales there since last Sep. Not quite following on the release plan. :)

**Comment by jelbourn** [comment]
@shb98 there's a new version of the radio button, currently in `@angular/material-experimental` based on MDC web that doesn't have this issue. We plan to promote this version to our stable `@angular/material` package sometime in 2022.

**Comment by amysorto** [comment]
When testing with NVDA and JAWS on Chrome for the new MDC based radio component coming in v15, this is fixed. I also tested the current non MDC based components (v14), and the right amount was read aloud as well.
